### ISSUE TITLE
Support $ref at path item object level

### ISSUE THREAD

**Issue by MauroB45** [drwpow/openapi-typescript#1107]
**Description**

Based on the openapi spec `$ref` is supported in the path-item-object config (see [spec](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#path-item-object)). When I tried generating the types for the following schema it does not generate the link correctly and the result object is empty.

``` yaml
openapi: "3.0"
info:
  title: test
  version: "1.0"
paths:
  /test:
    $ref: './_paths-test.yaml'
...
``` 

**Proposal**

Support `$ref` at the path item object level instead of the [operation object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#operation-object) where it is not supported (we can keep it for backward compatibility purposes).

**Checklist**

- [ ] I’m willing to open a PR for this (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/CONTRIBUTING.md))

**Comment by drwpow** [comment]
This is a definite bug and should be fixed. Not supporting valid `$ref`s isn’t an intentional decision.

**Comment by drwpow** [comment]
Released in `v6.3.8` 🎉
